### DISCUSSION TITLE
Any performance downsides to using the static binary?

### DISCUSSION THREAD

**Discussion by binaryfire** [php/frankenphp#503] [Q&A]
Hi all,

Does anyone know if there are performance downsides to using the static binary? Do things like JIT still work? If not, I might switch to the Docker image.

Cheers

**Comment by withinboredom** [comment]
Shared libraries are generally faster to startup and better memory usage (loading the library once vs. each time you start the binary) and on disk. Once it's up and running, the performance characteristics should be about the same. That being said, there may be other differences when it comes to Opcache and JIT.

I haven't really used the standalone mode, so I have very little experience with it. The best thing to do would be to profile your application with both and choose the one with the best trade-offs for your goals.

There's a good ops story for standalone mode (ease of deployment) that might make any speed differences not matter to you. Or the other way around. This will be different for everyone.

**Reply by binaryfire** [reply]
@withinboredom Thanks for the insights! Ease of deployment isn't an issue for me so my main concern is performance. I've just added a build to my pipeline using the `dunglas/frankenphp` image as a base. Haven't done any benchmarking yet but the memory usage seems lower at first glance. This also lets me just install the extensions I need, which probably helps memory consumption too.
